src/impls/std_collections/hashmap.rs

```diff
@@ -1,14 +1,19 @@
 use crate::{
-	Clear, Collection, CollectionMut, CollectionRef, Get, GetKeyValue, GetMut, Iter, Keyed,
-	KeyedRef, Len, MapInsert, MapIter, MapIterMut, Remove,
+	Clear, Collection, CollectionMut, CollectionRef, Entry, EntryApi, Get, GetKeyValue, GetMut,
+	Iter, KeyVacantEntry, Keyed, KeyedRef, Len, MapInsert, MapIter, MapIterMut, OccupiedEntry,
+	Remove, VacantEntry,
+};
+use std::{
+	borrow::Borrow,
+	collections::{hash_map, HashMap},
+	hash::{BuildHasher, Hash},
 };
-use std::{borrow::Borrow, collections::HashMap, hash::Hash};
 
-impl<K, V> Collection for HashMap<K, V> {
+impl<K, V, S: BuildHasher> Collection for HashMap<K, V, S> {
```

Comment:
Please make a separate PR with everything regarding `BuildHasher`. This PR should only be about the Entry API.